### PR TITLE
mcux: Build SNVS driver

### PR DESCRIPTION
**Pull request by pdgendt**
Builds the i.MX RT SNVS LP/HP drivers when the corresponding counter shim driver is enabled.

Required for https://github.com/zephyrproject-rtos/zephyr/pull/33150 to work

### CODE REVIEW

File: mcux/drivers/imx/CMakeLists.txt [hunk 1]
```diff
@@ -35,6 +35,8 @@ zephyr_library_sources_ifdef(CONFIG_UART_MCUX_LPUART	fsl_lpuart.c)
 zephyr_library_sources_ifdef(CONFIG_VIDEO_MCUX_CSI	fsl_csi.c)
 zephyr_library_sources_ifdef(CONFIG_WDT_MCUX_IMX_WDOG    fsl_wdog.c)
 zephyr_library_sources_ifdef(CONFIG_CAN_MCUX_FLEXCAN	fsl_flexcan.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER_MCUX_SNVS	fsl_snvs_hp.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER_MCUX_SNVS_SRTC	fsl_snvs_lp.c)
```
Comment (MaureenHelm) [inline mcux/drivers/imx/CMakeLists.txt:39]: Please move these up to preserve alpha order by CONFIG

**Review by MaureenHelm — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
@pdgendt Would you be able to address the review comment and also help resolve the merge conflict.

**Review by mmahadevan108 — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@MaureenHelm, any additional comments?

**Comment by mmahadevan108** [comment]
@dleach02 , can you help take a look at this

**Review by dleach02 — APPROVED** [review]

**Pull request merged**